flink-runtime/src/main/java/org/apache/flink/runtime/clusterframework/TaskExecutorResourceSpecBuilder.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.clusterframework;
+
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.configuration.MemorySize;
+import org.apache.flink.configuration.TaskManagerOptions;
+
+import static org.apache.flink.util.Preconditions.checkNotNull;
+
+/**
+ * Builder for {@link TaskExecutorResourceSpec}.
+ */
+public class TaskExecutorResourceSpecBuilder {
+
+	private final Configuration configuration;
+
+	private MemorySize totalProcessMemory = null;
+
+	private TaskExecutorResourceSpecBuilder(final Configuration configuration) {
+		this.configuration = new Configuration(checkNotNull(configuration));
+	}
+
+	public static TaskExecutorResourceSpecBuilder newBuilder(final Configuration configuration) {
+		return new TaskExecutorResourceSpecBuilder(configuration);
+	}
+
+	public TaskExecutorResourceSpecBuilder withTotalProcessMemory(MemorySize totalProcessMemory) {
+		this.totalProcessMemory = totalProcessMemory;
+		return this;
+	}
+
+	public TaskExecutorResourceSpec build() {
+		if (totalProcessMemory != null) {
+			configuration.setString(TaskManagerOptions.TOTAL_PROCESS_MEMORY, totalProcessMemory.toString());
```

Comment:
`configuration` is duplicated when constructing the builder. There's no need to always create a new `configuration` instance whenever the cpu / total process memory is updated.

Agree that we can change modify the configuration directly instead of keeping `totalProcessMemory`. I'll update it.

Reply:
BTW, do we want to make `changeConfigurationOption` public?
I think we can leave it private, since we do not have any use cases at the moment.